### ISSUE TITLE
Integration of GitHub Copilot Chat into Zed Code Editor

### ISSUE THREAD

**Issue by Walter-Hartwell-White** [zed-industries/zed#13092]
### Check for existing issues

- [ ] Completed

### Describe the feature

I would like to request the integration of GitHub Copilot Chat into the Zed code editor. GitHub Copilot Chat is an AI-powered tool that provides real-time code suggestions, explanations, and can assist with debugging directly within the editor. Integrating this feature into Zed would significantly enhance the user experience by providing developers with powerful coding assistance and seamless access to AI-driven insights. This feature would make Zed not only a robust code editor but also a smart development companion that can improve productivity and code quality.

This feature request is generated through AI but the emotions regarding this feature request is pure. Kindly consider my request because there are many devs like me who are student and not able to pay for the OpenAI API Key.
Thank You. 😭🙏🏻.

**Comment by SomeoneToIgnore** [comment]
@Walter-Hartwell-White , I find your behavior quite confusing.

TL;DR Stop spamming, this is a warning.

-------

You have to be clearly aware of the https://github.com/zed-industries/zed/issues/4673 issue as the one who left the comment there: https://github.com/zed-industries/zed/issues/4673#issuecomment-2167629845

Both that comment and this issue are very concerning to me, as you seem to fully ignore what's written prior, for example, the comment right above yours: https://github.com/zed-industries/zed/issues/4673#issuecomment-2167576056

To quote it:

```
Once again. CoPilot Chat depends on GitHub support. When Zed starts supporting the possibility of this type of integration in extensions, we will still have to count on a step from GitHub. Both the extensions for VSCode and JetBrains IDEs are OFFICIAL extensions - it is not possible to integrate with Chat via API (yet) so we have to count on the good heart of GitHub.

I think the Zed team knows perfectly well how important this functionality is for users (taking into account that Zed already supports a lot of LLM models), so don't spam please.
```

Nothing that was written on top by you (or LLM you've used) matters.
The only thing that will — a proper PR with the integration.

Hence, I again very strongly warn you against spamming on the topic more.

I will close this issue and minimize the comment aforementioned.
